StackExchange.Profiling/Timing.cs

```diff
@@ -81,21 +81,77 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public List<Timing> Children
```

Comment:
i think using IImmutableList for _children is an interesting idea. That would have indicated to the caller that the collection is immutable and make accessing _children naturally thread safe. 

I think adding children might be a bit more expensive since the tree is going to be built underneath but perhaps the downside is not that high, do you think we should make the change? I guess we (nor miniprofiler v4) are not going to support .net below 4.5 (where Immutable would not be available)